src/testrender/shading.cpp

```diff
@@ -30,157 +30,217 @@ enum ClosureIDs {
-                                                  CLOSURE_FLOAT_PARAM (ReflectionParams, eta),
-                                                  CLOSURE_FINISH_PARAM(ReflectionParams) } },
-        { "refraction" , REFRACTION_ID,         { CLOSURE_VECTOR_PARAM(RefractionParams, N),
-                                                  CLOSURE_FLOAT_PARAM (RefractionParams, eta),
-                                                  CLOSURE_FINISH_PARAM(RefractionParams) } },
-        { "transparent", TRANSPARENT_ID,        { CLOSURE_FINISH_PARAM(EmptyParams) } },
+        { "emission", EMISSION_ID, { CLOSURE_FINISH_PARAM(EmptyParams) } },
+        { "background", BACKGROUND_ID, { CLOSURE_FINISH_PARAM(EmptyParams) } },
+        { "diffuse",
+          DIFFUSE_ID,
+          { CLOSURE_VECTOR_PARAM(DiffuseParams, N),
+            CLOSURE_FINISH_PARAM(DiffuseParams) } },
+        { "oren_nayar",
+          OREN_NAYAR_ID,
+          { CLOSURE_VECTOR_PARAM(OrenNayarParams, N),
+            CLOSURE_FLOAT_PARAM(OrenNayarParams, sigma),
+            CLOSURE_FINISH_PARAM(OrenNayarParams) } },
+        { "translucent",
+          TRANSLUCENT_ID,
+          { CLOSURE_VECTOR_PARAM(DiffuseParams, N),
+            CLOSURE_FINISH_PARAM(DiffuseParams) } },
+        { "phong",
+          PHONG_ID,
+          { CLOSURE_VECTOR_PARAM(PhongParams, N),
+            CLOSURE_FLOAT_PARAM(PhongParams, exponent),
+            CLOSURE_FINISH_PARAM(PhongParams) } },
+        { "ward",
+          WARD_ID,
+          { CLOSURE_VECTOR_PARAM(WardParams, N),
+            CLOSURE_VECTOR_PARAM(WardParams, T),
+            CLOSURE_FLOAT_PARAM(WardParams, ax),
+            CLOSURE_FLOAT_PARAM(WardParams, ay),
+            CLOSURE_FINISH_PARAM(WardParams) } },
+        { "microfacet",
+          MICROFACET_ID,
+          { CLOSURE_STRING_PARAM(MicrofacetParams, dist),
+            CLOSURE_VECTOR_PARAM(MicrofacetParams, N),
+            CLOSURE_VECTOR_PARAM(MicrofacetParams, U),
+            CLOSURE_FLOAT_PARAM(MicrofacetParams, xalpha),
+            CLOSURE_FLOAT_PARAM(MicrofacetParams, yalpha),
+            CLOSURE_FLOAT_PARAM(MicrofacetParams, eta),
+            CLOSURE_INT_PARAM(MicrofacetParams, refract),
+            CLOSURE_FINISH_PARAM(MicrofacetParams) } },
+        { "reflection",
+          REFLECTION_ID,
+          { CLOSURE_VECTOR_PARAM(ReflectionParams, N),
+            CLOSURE_FINISH_PARAM(ReflectionParams) } },
+        { "reflection",
+          FRESNEL_REFLECTION_ID,
+          { CLOSURE_VECTOR_PARAM(ReflectionParams, N),
+            CLOSURE_FLOAT_PARAM(ReflectionParams, eta),
+            CLOSURE_FINISH_PARAM(ReflectionParams) } },
+        { "refraction",
+          REFRACTION_ID,
+          { CLOSURE_VECTOR_PARAM(RefractionParams, N),
+            CLOSURE_FLOAT_PARAM(RefractionParams, eta),
+            CLOSURE_FINISH_PARAM(RefractionParams) } },
+        { "transparent", TRANSPARENT_ID, { CLOSURE_FINISH_PARAM(EmptyParams) } },
         // mark end of the array
```

Comment:
This was more readable prior to clang-format.

Reply:
Not going to bother reverting because this is on the verge of being rewritten anyway as we synchronize all the closures with MaterialX